src/Table.js

```diff
@@ -617,11 +617,15 @@ export class Table {
     /*
         Convert items into a map of items by model type
     */
-    groupByType(items) {
+    groupByType(items, params={}) {
         let result = {}
         for (let item of items) {
             let type = item[this.typeField] || '_unknown'
             let list = result[type] = result[type] || []
+            let model = this.schema.models[type]
+            if (Object.keys(params).length && model && model !== this.schema.uniqueModel) {
+                item = model.transformReadItem('get', item, {}, params)
```

Comment:
I think the items have already been transformed, so this will be calling transformReadItem twice?

Reply:
The issue is that to be able to use groupby the items need to have the hidden attributes because it uses _type, I wanted to have a way to remove those hidden attributes after the group by, is there a clear way to do this?

Reply:
Understand.

So having params.hidden == true as you've got works fine.

You then want to copy the iteration logic from transformReadItem to pick out the non-hidden properties.

Reply:
Did you want to have a go at this, or do you want us to do it?

Reply:
Hey @mobsense sorry I did not respond, I will get something ready soon, one question I have should I still use the `OneParams` type here or should I just add something like 
```
groupByType(items, removeHidden=false) 
```

This way whoever uses this would know that the other props are not available ?

Reply:
Definitely use a map rather than a single option. That way we can provide for future options.

My preference is to use OneParams. The Model APIs use OneParams and certain APIs only support a sub-set of options as defined by the doc. This is a case where more is less. TS can get very complex where every API has a different params type. So the OneParams is a reasonable medium where the total set of options for APIs are defined.

Reply:
Got it 👍

Reply:
@mobsense let me know if [this](https://github.com/sensedeep/dynamodb-onetable/pull/171/commits/c474482623468b9da51bd716179fef719e29a25a#diff-a44f85a07e6bd8ef01c1de04a73f6db1da817c9bb48d523071f0f0f63ac20120R627) is what you meant.